client/windows/CCastleInterface.cpp

```diff
@@ -847,6 +847,17 @@ void CCastleBuildings::enterFountain(const BuildingID & building, BuildingSubID:
 	std::vector<std::shared_ptr<CComponent>> comps(1, std::make_shared<CComponent>(CComponent::building,town->subID, building));
 	std::string descr = town->town->buildings.find(building)->second->Description();
 
+	auto message677 = CGI->generaltexth->allTexts[677];
+	auto message678 = CGI->generaltexth->allTexts[678];
+	const auto pond = "pond";
+
+	if(this->town->town->faction->index != ETownType::RAMPART)
+	{
+		auto buildingName = town->town->getSpecialBuilding(subID)->Name();
+		boost::algorithm::replace_first(message677, pond, buildingName);

```

Comment:
This will work only with English localization. We need custom messages for custom buildings.